huggingface_sb3/push_to_hub.py

```diff
@@ -7,30 +7,309 @@
+    :param video_length: length of the video (in timesteps)
+    :param is_deterministic: use deterministic or stochastic actions
+    :param repo_local_path: path of the local repository
+    """
+    # Step 1: Create the VecVideoRecorder
+    env = VecVideoRecorder(
+        eval_env,
+        "./",  # Temporary video folder
+        record_video_trigger=lambda x: x == 0,
+        video_length=video_length,
+        name_prefix="",
+    )
+
+    obs = env.reset()
+    env.reset()
+    try:
+        for _ in range(video_length + 1):
+            action, _ = model.predict(obs, deterministic=is_deterministic)
+            obs, _, _, _ = env.step(action)
+            # env.render()
+
+        # Save the video
+        env.close()
+
+        # Rename the video
+        os.rename(env.video_recorder.path, "test.mp4")
+
+        # Convert the video with x264 codec
+        inp = "./test.mp4"
+        out = "replay.mp4"
+        os.system(f"ffmpeg -y -i {inp} -vcodec h264 {out}".format(inp, out))
+        """
+        call([ffmpeg,
+              '-i', "./replay.mp4",
+              '-vcodec', h264,
+              replay.mp4])
+        """
+        # Move the video
+        shutil.move(os.path.join("./", "replay.mp4"), os.path.join(repo_local_path, "replay.mp4"))
+    except KeyboardInterrupt:
+        pass
+    except:
+        # Add a message for video
+        print("We are unable to generate a replay of your agent")
+        print("Please send a message to thomas.simonini@huggingface.co")
+
+
+def select_tags(env_id):
+    """
+    Define the tags for the model card
+    :param env_id: name of the environment
+    """
+
+    model_card = f"""
+---
+      tags:
+      - {env_id}
+      - deep-reinforcement-learning
+      - reinforcement-learning
+      - stable-baselines3
+---
+    """
+    return model_card
+
+
+def _generate_model_card(model_name, env_id, mean_reward, std_reward):
     """
-    Creates a model card for the repository.
-    :param repo_dir:
+    Generate the model card for the Hub
+    :param model_name: name of the model
+    :env_id: name of the environment
+    :mean_reward: mean reward of the agent
+    :std_reward: standard deviation of the mean reward of the agent
+    """
+    # Step 1: Select the tags
+    model_card = select_tags(env_id)
+
+    # Step 2: Generate the model card
+    model_card += f"""
+  # **{model_name}** Agent playing **{env_id}**
+  This is a trained model of a **{model_name}** agent playing **{env_id}** using the [stable-baselines3 library](https://github.com/DLR-RM/stable-baselines3).
+  ## Evaluation Results
+  """
+
+    model_card += f"""
+  mean_reward={mean_reward:.2f} +/- {std_reward}
+  """
+
+    model_card += """
+  ## Usage (with Stable-baselines3)
+  TODO: Add your code
+  """
+
+    return model_card
+
+
+def _create_model_card(repo_dir: Path, generated_model_card):
+    """Creates a model card for the repository.
+    TODO: Add metrics to model-index
```

Comment:
If possible I would already do this in this PR. You can use https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/repocard.py `metadata_save` and `metadata_eval_result` that should make your life easier than handling all the stuff yourself.